Segfault while reloading configuration

While investigating the issue #1354 I hit a pod with an error due to the following segfault

```
{"level":"info","ts":"2023-06-30T12:35:15Z","msg":"Requesting configuration reload","logging_pod":"cluster-pg-replication-slot-3","pgdata":"/var/lib/postgresql/data/pgdata"}
{"level":"info","ts":"2023-06-30T12:35:15Z","logger":"pg_ctl","msg":"server signaled","pipe":"stdout","logging_pod":"cluster-pg-replication-slot-3"}
{"level":"info","ts":"2023-06-30T12:35:15Z","logger":"postgres","msg":"record","logging_pod":"cluster-pg-replication-slot-3","record":{"log_time":"2023-06-30 12:35:15.881 UTC","process_id":"22","session_id":"649ecbf9.16","session_line_num":"7","session_start_time":"2023-06-30 12:35:05 UTC","transaction_id":"0","error_severity":"LOG","sql_state_code":"00000","message":"received SIGHUP, reloading configuration files","backend_type":"postmaster","query_id":"0"}}
{"level":"info","ts":"2023-06-30T12:35:15Z","logger":"postgres","msg":"record","logging_pod":"cluster-pg-replication-slot-3","record":{"log_time":"2023-06-30 12:35:15.883 UTC","process_id":"22","session_id":"649ecbf9.16","session_line_num":"8","session_start_time":"2023-06-30 12:35:05 UTC","transaction_id":"0","error_severity":"LOG","sql_state_code":"00000","message":"parameter \"primary_slot_name\" changed to \"_cnpg_cluster_pg_replication_slot_3\"","backend_type":"postmaster","query_id":"0"}}
{"level":"info","ts":"2023-06-30T12:35:15Z","msg":"Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference","controller":"cluster","controllerGroup":"postgresql.cnpg.io","controllerKind":"Cluster","Cluster":{"name":"cluster-pg-replication-slot","namespace":"replication-slot-e2e-9152"},"namespace":"replication-slot-e2e-9152","name":"cluster-pg-replication-slot","reconcileID":"001932d3-f9d3-4496-b38a-1dcbe21f611e"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x42 pc=0x1a5d03b]

goroutine 112 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	/home/zeus/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:115 +0x1fa
panic({0x1c9c8a0, 0x316e1e0})
	/usr/local/go/src/runtime/panic.go:884 +0x213
github.com/cloudnative-pg/cloudnative-pg/internal/management/controller.(*InstanceReconciler).processConfigReloadAndManageRestart(0xc0004e49c0, {0x225e370, 0xc00030fef0}, 0xc00078ce00)
	internal/management/controller/instance_controller.go:903 +0x7b
github.com/cloudnative-pg/cloudnative-pg/internal/management/controller.(*InstanceReconciler).Reconcile(0xc0004e49c0, {0x225e370?, 0xc00030fe00?}, {{{0xc00030fe00?, 0x0?}, {0xc000aa9d20?, 0x40e1a7?}}})
	internal/management/controller/instance_controller.go:217 +0x80c
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x225e370?, {0x225e370?, 0xc00030fe00?}, {{{0xc0005d8500?, 0x1bb2b80?}, {0xc0005d84e0?, 0x224a630?}}})
	/home/zeus/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:118 +0xc8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0004f6320, {0x225e2c8, 0xc000340b90}, {0x1d6f340?, 0xc000aa1840?})
	/home/zeus/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:314 +0x377
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0004f6320, {0x225e2c8, 0xc000340b90})
	/home/zeus/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:265 +0x1d9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	/home/zeus/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:226 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/home/zeus/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:222 +0x587
```

This happens in the tests `Replication Slot` and the pod `cluster-pg-replication-slot-3` is the first one, it's produced after there's a reload for the replication slots

--- a/internal/management/controller/instance_controller.go
+++ b/internal/management/controller/instance_controller.go
@@ -900,7 +900,7 @@ func (r *InstanceReconciler) processConfigReloadAndManageRestart(ctx context.Con
 		return err
 	}
 
-	if !status.PendingRestart {
+	if status == nil || !status.PendingRestart {
 		// Everything fine
 		return nil
 	}

